sensor/mcp9808/mcp9808.go

```diff
@@ -0,0 +1,484 @@
+func (d *MCP9808) AlertSelect() bool {
+	return d.readConfigValue(configAlertSelect)
+}
+
+// SetAlertSelect - Alert Output Select bit
+// 0 (false) = Alert output for TUPPER, TLOWER and TCRIT (power-up default)
+// 1 (true) = TA > TCRIT only (TUPPER and TLOWER temperature boundaries are disabled)
+// When the Alarm Window Lock bit is set, this bit cannot be altered until unlocked (bit 6).
+// This bit can be programmed in Shutdown mode, but the Alert output will not assert or deassert.
+func (d *MCP9808) SetAlertSelect(set bool) {
+	d.flipConfigBit(configAlertSelect, set)
+}
+
+// AlertPolarity - Alert Output Polarity bit
+// 0 (false) = Active-low (power-up default; pull-up resistor required)
+// 1 (true) = Active-high
+func (d *MCP9808) AlertPolarity() bool {
+	return d.readConfigValue(configAlertPolarity)
+}
+
+// SetAlertPolarity - Alert Output Polarity bit
+// 0 (false) = Active-low (power-up default; pull-up resistor required)
+// 1 (true) = Active-high
+// This bit cannot be altered when either of the Lock bits are set (bit 6 and bit 7).
+// This bit can be programmed in Shutdown mode, but the Alert output will not assert or deassert.
+func (d *MCP9808) SetAlertPolarity(set bool) {
+	d.flipConfigBit(configAlertPolarity, set)
+}
+
+// AlertMode - Alert Output Mode bit
+// 0 (false) = Comparator output (power-up default)
+// 1 (true) = Interrupt output
+func (d *MCP9808) AlertMode() bool {
+	return d.readConfigValue(configAlertMode)
+}
+
+// SetAlertMode - Alert Output Mode bit
+// 0 (false) = Comparator output (power-up default)
+// 1 (true) = Interrupt output
+// This bit cannot be altered when either of the Lock bits are set (bit 6 and bit 7).
+// This bit can be programmed in Shutdown mode, but the Alert output will not assert or deassert.
+func (d *MCP9808) SetAlertMode(set bool) {
+	d.flipConfigBit(configAlertMode, set)
+}
+
+// Temperature contains the ambient temperature along with alert values.
+type Temperature struct {
+	CelsiusDeg                            float64
+	AboveCritical, AboveUpper, BelowLower bool
+}
+
+// AmbientTemp reads the current sensor value along with the flags denoting what boundaries the
+// current temperature exceeds.
+func (d *MCP9808) AmbientTemp() (*Temperature, error) {
+	temp, err := d.Bus.ReadWordFromReg(address, regAmbientTemp)
+	if err != nil {
+		return nil, err
+	}
+
+	tempC := temp &^ 0xF000
+
+	wholeNum := float64(tempC&^0xF) / 16.0
+	fraction := tempC &^ 0xFF0
+
+	tempResult := &Temperature{
+		AboveCritical: !(temp&(1<<15) == 0),
+		AboveUpper:    !(temp&(1<<14) == 0),
+		BelowLower:    !(temp&(1<<13) == 0)}
+	tempResult.CelsiusDeg = wholeNum + (float64(fraction) / 10000.0)
+	if !(temp&(1<<12) == 0) { // read sign bit
+		tempResult.CelsiusDeg *= -1
+	}
+	return tempResult, nil
+}
+
+// readTempC reads from the ambient temperature register and returns the current temperature value in celsius
+func (d *MCP9808) readTempC(reg byte) (float64, error) {
```

Comment:
This has a lot of code duplicated with AmbientTemp, why not factor out the common code? E.g. AmbientTemp could call readTempC and then add the info in the extra bits...